src/webots/nodes/WbJointDevice.cpp

```diff
@@ -48,8 +48,8 @@ void WbJointDevice::init() {
   mRobot = NULL;
 }
 
-void WbJointDevice::postFinalize() {
-  WbBaseNode::postFinalize();
+void WbJointDevice::preFinalize() {
+  WbBaseNode::preFinalize();
   // Cache position index
   const WbField *const f = parentField(true);
```

Comment:
Just for info: usually, all the instructions that depends on the parent field/node hierarchy are usually executed in the `postFinalize` method because of this call:
https://github.com/cyberbotics/webots/blob/c0ce8250876997125c4decbeed48bf0f98e39e0b/src/webots/nodes/WbBaseNode.cpp#L101

In this particular case, the `WbNode::parentField()` result should not make use of the field parent value.

Reply:
Then if mPositionIndex should be set only in the `postFinalize` it kinda defies the purpose of the rework entirely. So the second option of simply moving the call of  `updateMinAndMaxPosition` to the `WbMotor::postFinalize` to fix the issue would've been better? If not, mPositionIndex will always be 1. (unless there's a better way of knowing which axis is being concerned)

Reply:
It should be checked and tested, but from a first look it seems fine to move this to the `preFinalize()` method because it doesn't use the `WbField::parentNode()` method that is set after the `preFinalize` step.

I just wanted to give a context to the choice of adding this code to the `postFinalize` method.

Reply:
Oh I see, thanks for the info. Well as far as I can tell it seems fine, neither this nor the derived ones use it. But just to be sure, it might be better if the reviewer resolves this conversation rather than me (assuming the tests don't break in the meantime).